[GOMUNGNAM] step-3 구현 (DB연동, 배포)

KOKEONHO opened the pull request:
* h2 사용하여 DB 연동했습니다
* 배포는 aws를 사용했습니다
* 호기심에 도메인을 구매하여 연결했습니다

www.gomungnamcafe.site

---
처음으로 JDBC를 사용해봤는데 생각보다 DB와 연결하는 과정이 재미있었습니다. 그리고 내가 연결한 뷰와 정보들이 DB에 저장되는게 신기했습니다.

KOKEONHO committed:
Refactor: build.gradle 코드 수정

h2 데이터베이스 사용과 연결을 위해 dependencies에 코드 추가

KOKEONHO committed:
Refactor: application.properties 코드 수정

h2 데이터베이스 연결을 위한 코드 추가

KOKEONHO committed:
Feat: ConnectionConst.java 생성

h2 데이터베이스와의 연결에 필요한 URL, USERNAME, PASSWORD를 상수로 선언한 ConnectionConst.java 생성

KOKEONHO committed:
Feat: DBConnectionUtility.java 생성

JDBC를 사용하여 실제 h2 데이터베이스와 연결을 수행하는 DBConnectionUtility.java 생성

KOKEONHO committed:
Feat: DBConnectionUtility.java 기능 추가

상수 처리한 URL, USERNAME, PASSWORD를 이용하여 Connection을 생성하는 기능 추가

KOKEONHO committed:
Refactor: Member.java 코드 수정

User가 SQL의 예약어인 관계로 기존의 User를 Member로 수정함에 따라 발생한 코드 수정

KOKEONHO committed:
Refactor: MemberController.java 코드 수정

User가 SQL의 예약어인 관계로 기존의 User를 Member로 수정함에 따라 발생한 코드 수정

KOKEONHO committed:
Refactor: MemberRepository.java 코드 수정

User가 SQL의 예약어인 관계로 기존의 User를 Member로 수정함에 따라 발생한 코드 수정

KOKEONHO committed:
Refactor: index.html 코드 수정

User가 SQL의 예약어인 관계로 기존의 User를 Member로 수정함에 따라 발생한 코드 수정

KOKEONHO committed:
Refactor: EOL 추가

파일의 끝에 EOL 추가

KOKEONHO committed:
Refactor: show.html 코드 수정

User가 SQL의 예약어인 관계로 기존의 User를 Member로 수정함에 따라 발생한 코드 수정

KOKEONHO committed:
Refactor: Member.java 코드 수정

DB와 연결하면서 불필요한 코드 주석 처리

KOKEONHO committed:
Refactor: MemberController.java 코드 수정

발생한 SQLException을 throw해주는 코드 추가 및 불필요한 코드 주석 처리

KOKEONHO committed:
Feat: MemberRepository.java 기능 추가

연결된 DB에 생성된 Member의 정보를 전달하는 기능 추가

KOKEONHO committed:
Test: MemberRepositoryTest.java 생성

연결된 DB에 생성된 Member가 문제 없이 저장되는지 확인하는 MemberRepositoryTest.java 생성

KOKEONHO committed:
Test: DBConnectionUtilityTest.java 생성

DB와 연결이 문제 없이 정상 작동하는지 확인하는 DBConnectionUtilityTest.java 생성

KOKEONHO committed:
Refactor: ddl.sql 코드 수정

기존 member_number로 설정해놨던 primary key를 member_id로 재할당

KOKEONHO committed:
Feat: build.gradle 코드 수정

테스트에서 Slf4j를 사용하기 위해 build.gradle에 lombok 추가

KOKEONHO committed:
Feat: MemberController.java 코드 수정

SQLException을 throw하는 코드 추가

KOKEONHO committed:
Feat: MemberRepository.java 기능 추가

Member의 ID를 통해 해당 Member의 정보를 불러오는 기능 추가

KOKEONHO committed:
Test: MemberRepositoryTest.java 기능 추가

Member의 추가와 ID를 통한 해당 Member의 정보 불러오는 기능 테스트 추가

KOKEONHO committed:
Feat: Member.java 기능 추가

기본 생성자 추가

KOKEONHO committed:
Refactor: MemberController.java 코드 수정

메서드 이름 변경으로 인한 코드 수정

KOKEONHO committed:
Feat: MemberRepository.java 기능 추가

Member의 정보를 수정할 때 비밀번호가 일치하면 정상적으로 수정을 하고 불일치 시에는 수정을 하지 않는 기능 추가

KOKEONHO committed:
Test: MemberRepositoryTest.java 기능 추가

회원 정보 업데이트 테스트 추가

KOKEONHO committed:
Refactor: MemberController.java 코드 수정

SQLException을 throw하는 코드 추가

KOKEONHO committed:
Feat: MemberRepository.java 기능 추가

모든 Member들의 정보를 List로 가져오는 기능 추가

KOKEONHO committed:
Test: MemberRepositoryTest.java 기가능 추가

모든 Member들의 정보를 조회하는 테스트 추가

KOKEONHO committed:
Refactor: application.properties 코드 수정

spring.datasource.url 코드 수정

KOKEONHO committed:
Refactor: ConnectionConst.java 코드 수정

URL 주소 수정

KOKEONHO committed:
Refactor: Member.java 코드 수정

DB에서 정보를 뽑아올 때 사용자의 번호를 가져올 수 있도록 userSequence 변수 다시 추가

KOKEONHO committed:
Feat: MemberRepository.java 코드 수정

DB에서 사용자의 정보를 조회할 때 primary key인 user_number를 userSequence에 저장할 수 있도록 코드 수정

KOKEONHO committed:
Feat: Article.java 기능 추가

게시글이 써진 시간을 저장하는 writtenTime을 객체 생성 시에 초기화하는 기능 추가

KOKEONHO committed:
Refactor: index.html 코드 수정

게시글이 써진 시간을 mapping하는 코드 수정

KOKEONHO committed:
Feat: ddl.sql 추가 작성

article 테이블 구조 추가 작성

KOKEONHO committed:
Refactor: ArticleController.java 코드 수정

SQLException을 throw하는 코드 추가

KOKEONHO committed:
Refactor: schema.sql 코드 수정

기존 ddl.sql이었던 이름을 schema.sql로 수정하고 DB 시작 시에 초기화 해주는 코드 추가

KOKEONHO committed:
Refactor: ArticleRepository.java 코드 수정

Article의 primary key인 번호로 해당 article을 상세 열람할 수 있는 기능 추가

KOKEONHO committed:
Feat: Article.java 기능 추가

게시글 작성 시간을 나타내는 writtenTime 추가

KOKEONHO committed:
Feat: application.properties 기능 추가

DB 시작 시 자동으로 초기화 및 초기화 파일 경로 지정 기능 추가

KOKEONHO committed:
Feat: application.properties 기능 추가

web-allow-others true로 설정

KOKEONHO committed:
Feat: data.sql 생성

DB에 자동적으로 테스트 데이터를 넣어주는 data.sql 생성

honux77 commented on build.gradle:
```diff
@@ -14,8 +14,15 @@ repositories {
+	runtimeOnly 'com.h2database:h2:2.1.214'
 	implementation 'org.springframework.boot:spring-boot-starter-web'
 	testImplementation 'org.springframework.boot:spring-boot-starter-test'
+	compileOnly 'org.projectlombok:lombok'
+	annotationProcessor 'org.projectlombok:lombok'
+
+	testCompileOnly 'org.projectlombok:lombok'
+	testAnnotationProcessor 'org.projectlombok:lombok'
```
롬복 의존성과 코드는 제거해 주세요. 안 쓰는 게 좋습니다.

KOKEONHO replied:
테스트에서 로그를 찍기 위해서 사용하고 있었는데 테스트에서도 사용하지 않는게 좋을까요 ?

honux77 commented on src/main/java/kr/codesqaud/cafe/controller/MemberController.java:
```diff
@@ -0,0 +1,79 @@
+        return "redirect:/users/list";
+    }
+
+//    /**
+//     * 테스트 용 데이터 추가
+//     */
+//    @PostConstruct
+//    public void init() throws SQLException {
+//        memberRepository.save(new Member("GOMUNGNAM", "247597", "고건호", "rhrjsgh97@gmail.com"));
+//        memberRepository.save(new Member("Roy", "123456", "이승로", "roy@gmail.com"));
+//        memberRepository.save(new Member("Birdie", "123456", "임동현", "birdie@gmail.com"));
+//        memberRepository.save(new Member("Hana", "1234", "왕하나", "hana@gmail.com"));
+//    }
```
불필요한 주석은 항상 제거해 주세요.

KOKEONHO replied:
불필요한 주석 다 제거했습니다

honux77 commented on src/main/java/kr/codesqaud/cafe/controller/ArticleController.java:
```diff
@@ -24,7 +25,7 @@ public ArticleController(ArticleRepository articleRepository) {
 
     @GetMapping("/")
-    public String articles(Model model) {
+    public String articles(Model model) throws SQLException {
```
```suggestion
    public String listArticles(Model model) throws SQLException {
```

KOKEONHO replied:
수정했습니다 !

honux77 reviewed (changes requested):
1. 커넥션을 직접 사용하지 않습니다. 스프링에서 제공해 주는 데이터소스를 활용해 주세요.
2. Spring JDBC template으로 수정해 주세요.

KOKEONHO committed:
Refactor: MemberController.java 코드 수정

불필요한 주석 삭제

KOKEONHO committed:
Refactor: ArticleRepository.java 코드 수정

불필요한 주석 삭제

KOKEONHO committed:
Refactor: MemberRepository.java 코드 수정

불필요한 주석 삭제

KOKEONHO committed:
Refactor: ArticleController.java 코드 수정

기존의 메서드 명이었던 articles에서 listArticles로 메서드 명 변경

KOKEONHO committed:
Feat: JdbcTemplateMemberRepository.java 생성

JDBC Template을 사용하여 DB와 연결시킬 JdbcTemplateMemberRepository.java 생성

KOKEONHO committed:
Feat: JdbcTemplateMemberRepository.java 기능 추가

RowMapper<Member>를 반환하는 memberRowMapper() 메서드 추가

KOKEONHO committed:
Feat: JdbcTemplateMemberRepository.java 기능 추가

사용자를 사용자의 ID로 찾을 수 있는 findById() 메서드 구현

KOKEONHO committed:
Refactor: JdbcTemplateMemberRepository.java 코드 수정

기존 JdbcTemplate을 사용하던 코드를 NamedParameterJdbcTemplate으로 수정

KOKEONHO committed:
Refactor: MemberController.java 코드 수정

기존 JdbcTemplate을 사용하던 코드를 NamedParameterJdbcTemplate으로 수정

KOKEONHO committed:
Refactor: MemberController.java 코드 수정

기존 jdbcTemplateMemberRepository를 변수명으로 사용하던 코드를 namedJdbcTemplateMemberRepository로 변경

KOKEONHO committed:
Feat: NamedJdbcTemplateMemberRepository.java 기능 추가

비밀번호가 불일치 시에 회원 정보를 수정하지 않는 기능 추가

KOKEONHO committed:
Feat: JdbcTemplateArticleRepository.java 생성

JDBC Template을 사용하여 DB와 연결시킬 JdbcTemplateArticleRepository.java 생성

KOKEONHO committed:
Rename: JdbcTemplateArticleRepository.java 파일명 수정

기존 JdbcTemplateArticleRepository.java였던 파일 이름을 NamedJdbcTemplateArticleRepository.java로 변경

KOKEONHO committed:
Refactor: ArticleController.java 코드 수정

새롭게 생성한 NamedJdbcTemplateArticleRepository로 기존 ArticleRepository 대체하는 코드 수정

KOKEONHO committed:
Feat: NamedJdbcTemplateArticleRepository.java 기능 추가

NamedJdbcTemplate을 사용하여 게시판의 기능들을 NamedJdbcTemplateArticleRepository.java에 추가

KOKEONHO committed:
Refactor: ArticleController.java 코드 수정

불필요한 주석 삭제

KOKEONHO committed:
Refactor: MemberController.java 코드 수정

불필요한 주석 삭제
